components/brave_welcome_ui/components/screens/searchBox.tsx

```diff
@@ -18,11 +19,64 @@ interface Props {
   index: number
   currentScreen: number
   onClick: () => void
+  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void
+  isDefaultSearchGoogle: boolean
 }
 
-export default class SearchEngineBox extends React.PureComponent<Props, {}> {
+interface State {
+  searchEngineSelected: boolean,
+  searchEngineIndex: number,
+  searchProviders: Array<any>
+}
+
+interface SearchEngineEntry {
+  canBeDefault: boolean,
+  canBeEdited: boolean,
+  canBeRemoved: boolean,
+  default: boolean,
+  displayName: string,
+  iconURL: string,
+  id: number,
+  isOmniboxExtension: boolean,
+  keyword: string,
+  modelIndex: number,
+  name: string,
+  url: string,
+  urlLocked: boolean
+}
+
+export default class SearchEngineBox extends React.PureComponent<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      searchEngineSelected: false,
+      searchEngineIndex: -1,
+      searchProviders: []
+    }
+
+    // @ts-ignore
+    window.cr.sendWithPromise('getSearchEnginesList').then(response => {
```

Comment:
Not sure how to properly expose this via typescript? so had to ts-ignore it

Method is defined in https://github.com/brave/brave-core/blob/c90dbb4a7166436b1dcb1b3eacbb6184efb8f6a4/components/brave_welcome_ui/brave_welcome.html#L10

Reply:
@imptrx could you create a helper method for this? also add a `catch` at the promise's end with a warning like "unable to get the search engines list" or something you think fits better

Reply:
also please type the method

Reply:
@cezaraugusto I am thinking it be cleaner if we move this method under the welcome page component's `initialize` method and pipe it through redux before populating it here. It looks like we are doing something along those lines in 'brave_adblock` - would that be a good reference point?

Reply:
sounds good 👍